dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/ClassDumper.java

```diff
@@ -416,6 +443,16 @@ static boolean classesInSamePackage(String classNameA, String classNameB) {
     return packageNameA.equals(packageNameB);
   }
 
+  /** Returns true if {@code childClass} is a subclass of {@code parentClass}. */
+  static boolean isClassSubClassOf(JavaClass childClass, JavaClass parentClass) {
+    for (JavaClass superClass : getClassAndSuperClasses(childClass)) {
```

Comment:
Might be simpler as a while or do while loop that operates while parent is not null.

Reply:
I just tried the idea, but as `childClass.getSuperClass()` may throw `ClassNotFoundException` (a checked exception), the logic did not become simpler.